### ISSUE TITLE
Debug instructions are not working for me

### ISSUE THREAD

**Issue by adnan360** [Phate6660/bcalc#5]
Congratulations on crossing off so many TODOs!

I've checked the readme and it has some commands on debugging:

```
$ ./bcalc [100]-[50]
final_array = [ 1 0 0 ] - [ 5 0 ]
number_array = 100 50
operation_array = -
calculation = 100-50
50
```

But when I execute the same, it gives me:

```
$ ./bcalc [100]-[50]
50
```

Maybe there are some parameters missing?

**Comment by Phate6660** [comment]
Yep, that's my bad. 
I must not have copied the command when pasting into the readme.
Sorry about that. Add `--debug` as an arg.

**Comment by adnan360** [comment]
No problem.

I think the debug arg has to be put after the input:

```
$ ./bcalc --debug 2+2
Only digits, parenthesis, square brackets, '*', '/', '+', and '-', are supported.
'7' is currently unsupported.
$ ./bcalc 2+2 --debug
final_array = 2 + 2
number_array[0] = 2
...
```

One question though, why is it showing "'7' is currently unsupported."?

**Comment by Phate6660** [comment]
It has to be the second argument.
The first argument is the actual math stuff.
I should probably specify that in the readme.

**Comment by Phate6660** [comment]
I'm actually not sure. I'll look into that because I'm curious to know myself how it got a 7 from anywhere.

**Comment by Phate6660** [comment]
Figured it out. The 7 is due to a misnamed variable.
My bad lol.

**Comment by adnan360** [comment]
I think it could be something else. It's probably getting the string length of the arg:

```
$ ./bcalc --help
Only digits, parenthesis, square brackets, '*', '/', '+', and '-', are supported.
'6' is currently unsupported.
$ ./bcalc -test
Only digits, parenthesis, square brackets, '*', '/', '+', and '-', are supported.
'5' is currently unsupported.
```

**Comment by adnan360** [comment]
> I should probably specify that in the readme.

That'd be great.

**Comment by Phate6660** [comment]
Yep, it is. I had renamed the variable in use (i) to char. But I forgot to change the var in the error message. I'm using i in a for loop to that iterates over the argument length.

**Comment by adnan360** [comment]
I see. Another bug solved then. :+1: 

Also this issue is fixed in 8aca861a108821f0f3b6ba198fe703c969fcf981. So closing.
